src/ExpressionEvaluator/Core/Source/ExpressionCompiler/EvaluationContextBase.cs

```diff
@@ -110,6 +110,8 @@ internal string GetErrorMessageAndMissingAssemblyIdentities(DiagnosticBag diagno
 
         internal abstract ImmutableArray<AssemblyIdentity> GetMissingAssemblyIdentities(Diagnostic diagnostic);
 
+        internal abstract bool HasReferenceToAny(ImmutableArray<AssemblyIdentity> identities);
```

Comment:
After talking to Chuck just now, I'm going to switch back to the fix I did for preview (create a HashSet on the side).  It's a more localized fix, and the additional memory overhead should be small.